packages/react-composites/src/composites/ChatComposite/adapter/FileSharingAdapter.ts

```diff
@@ -0,0 +1,158 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import produce from 'immer';
+import { FileMetadata, FileSharingMetadata, ObservableFileUpload, FileUploadState } from '../file-sharing';
+import { ChatContext } from './AzureCommunicationChatAdapter';
+import { ChatAdapterState } from './ChatAdapter';
+
+/**
+ * A record containing {@link FileUploadState} mapped to unique ids.
+ * @beta
+ */
+export type FileSharingUiState = Record<string, FileUploadState>;
+
+/**
+ * @beta
+ */
+export interface FileUploadAdapter {
```

Comment:
I think edge also lost some of my comments.

Possibly repeating this one: We're currently using "file upload" and "file sharing" inconsistently:

* This file is `FileSharingAdapter.ts`
* This interface is `FileUploadAdapter`, and is implemented by `FileSharingAdapter`
* The state is stored in `FileSharingUiState` that uses `FileUploadState`.

I think all of this API surface is for file upload only, so I'd recommend just sticking with "file upload" consistently in naming.

Reply:
Haha yes, I agree.
Just realised this. Lemme wreak havoc on the code once again.

Reply:
Check it out. Renamed everything to be consistent